openbb_platform/providers/tradier/openbb_tradier/models/options_chains.py

```diff
@@ -275,6 +211,51 @@ def transform_data(
         query: TradierOptionsChainsQueryParams,
         data: List[Dict],
         **kwargs: Any,
-    ) -> List[TradierOptionsChainsData]:
+    ) -> TradierOptionsChainsData:
         """Transform and validate the data."""
-        return [TradierOptionsChainsData.model_validate(d) for d in data]
+        # pylint: disable = import-outside-toplevel
+        from dateutil.parser import parse
+        from numpy import nan
+        from openbb_core.provider.utils.helpers import safe_fromtimestamp
+        from pandas import DataFrame
+        from pytz import timezone
+
+        def df_apply_dates(v):
+            """Validate the dates."""
+            if v != 0 and v is not None and isinstance(v, int):
+                v = int(v) / 1000  # milliseconds to seconds
+                v = safe_fromtimestamp(v)
+                v = v.replace(microsecond=0)
+                v = v.astimezone(timezone("America/New_York"))
+                return v
+            if v is not None and isinstance(v, str):
+                v = parse(v)
+                v = v.replace(microsecond=0, tzinfo=timezone("UTC"))
+                v = v.astimezone(timezone("America/New_York"))
+                return v
+            return None
+
+        def map_exchange(v):
+            """Map the exchange from a code to a name."""
+            return (
+                OPTIONS_EXCHANGES.get(v)
+                if v in OPTIONS_EXCHANGES
+                else (
+                    STOCK_EXCHANGES.get(v) if v in STOCK_EXCHANGES else v if v else None
+                )
+            )
+
+        output = DataFrame(data)
+        for col in output:
+            if col not in ["dte", "open_interest", "volume"]:
+                output[col] = output[col].replace({0: None})
+            if col in ["bid_date", "ask_date", "trade_date", "updated_at"]:
+                output[col] = output[col].apply(df_apply_dates)
+            if col == "change_percentage":
```

Comment:
You can do an `elif` on this one I believe.

Reply:
Done